### PR TITLE
[TimesFMv1] fix variance calculation

### PR DESCRIPTION
**Pull request by kashif**
1. Masked variance calculation (lines 95-107): Changed from the numerically unstable E[X²] - E[X]² formula  to the stable centered formula E[(X-μ)²]
  2. Sigma clamping (line 609): Changed from torch.where(sigma < tolerance, 1.0, sigma) to torch.clamp(sigma,   min=tolerance)

by: https://github.com/huggingface/transformers/pull/42099 @AnMakc

### CODE REVIEW

**Comment by AnMakc** [comment]
@kashif 
Thanks for bringing it here.
Could you also fix the jax version, it has the same issue: https://github.com/google-research/timesfm/blob/bf88c5dc88f6275b7071ebcd9051406c11435c0c/v1/src/timesfm/patched_decoder.py#L199

<br><br>
Btw, this potentially affected model pre-training as well.
